### PR TITLE
feat(prefer-action-creator-in-dispatch): add rule

### LINKED ISSUE #238: A rule to enforce `.dispatch(action())`
## Information about the new rule
As mentioned in https://github.com/timdeschryver/eslint-plugin-ngrx/pull/211#discussion_r712270311, we could create a rule for this.

Do you have any name in mind?

## A snippet of incorrect code
```ts
this.store$.dispatch({ type: 'myAction' });
this.store$.dispatch(new MyAction());

```

## A snippet of correct code
```ts
this.store$.dispatch(action());
```
## I would be willing to submit a PR for the docs :heart:

[x] Yes (Assistance is provided if you need help submitting a pull request)
[ ] No

### PR DESCRIPTION
**Pull request by rafaelss95**
Closes #238.

### CODE REVIEW

**Comment by rafaelss95** [comment]
Converting to draft until we reach consensus on name...

File: src/rules/store/prefer-action-creator-in-dispatch.ts [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+
+export const messageId = 'preferActionCreatorInDispatch'
+export type MessageIds = typeof messageId
+
+type Options = []
+
+export default ESLintUtils.RuleCreator(docsUrl)<Options, MessageIds>({
+  name: path.parse(__filename).name,
+  meta: {
+    type: 'problem',
```
Comment (timdeschryver) [inline src/rules/store/prefer-action-creator-in-dispatch.ts:14]: Should we use `suggestion` for all of our `prefer-` rules 🤔

Comment (rafaelss95) [reply src/rules/store/prefer-action-creator-in-dispatch.ts]: Addressed 💯. Your comment made me realize that today we have `prefer-concat-latest-from` as `type: 'suggestion'`, but shouldn't it be `problem` instead, as misusing `withLatestFrom` can lead to some problems?

Comment (timdeschryver) [reply src/rules/store/prefer-action-creator-in-dispatch.ts]: You're also right! That should be `problem`.
This is hard 😅

**Review by timdeschryver — APPROVED** [review]

**Pull request merged**